### PR TITLE
frontend-tools: Make links in script description clickable

### PR DESCRIPTION
**Pull request by WizardCM**
# Description

QLabel supports super basic HTML, allowing for formatting for readability, and clickable links. This ensures clickable links can actually be clicked, rather than users having to Right Click -> "Copy Link Location".

image

Additionally, this code ensures that the clickable link is pointed at a webpage, to avoid opening other protocols, and includes a confirmation dialog to show the user where they're going.

### Motivation and Context

OBS already turns anchor tags into a link, but clicking it does nothing, as Qt by default will try to open it internally.

This also makes the description more useful: the script can provide an easy link to a GitHub repo (for updates), the developer's website, or to a dependency that is required for the script to function.

### How Has This Been Tested?

Modified the description of `countdown.lua`, the provided countdown script, to use html with an anchor tag.
```lua
function script_description()
	return "Sets a text source to act as a countdown timer when the source is active.<br><br>Made by Jim. Testing clickable links <a href=\"http://obsproject.com\">in OBS</a>."
end
```
**Note**: the script's description originally used `\n` for newlines, which disables HTML parsing

### Types of changes

 - New feature (non-breaking change which adds functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: UI/frontend-plugins/frontend-tools/data/locale/en-US.ini [hunk 1]
```diff
@@ -38,6 +38,11 @@ PythonSettings.PythonInstallPath64bit="Python Install Path (64bit)"
 PythonSettings.BrowsePythonPath="Browse Python Path"
 ScriptLogWindow="Script Log"
 Description="Description"
+ScriptDescriptionLink.Title="Open URL"
+ScriptDescriptionLink.Text="You have clicked a link. This will open a webpage in your default web browser. Continue?"
```
Comment (JohannMG) [inline UI/frontend-plugins/frontend-tools/data/locale/en-US.ini:42]:
```suggestion
ScriptDescriptionLink.Text="Open this link in your default browser?"
```

File: UI/frontend-plugins/frontend-tools/scripts.cpp [hunk 2]
```diff
@@ -470,6 +474,37 @@ void ScriptsTool::on_defaults_clicked()
 		item->data(Qt::UserRole).toString().toUtf8().constData());
 }
 
+void ScriptsTool::on_description_linkActivated(const QString &link)
+{
+	if (link.left(7).compare("http://", Qt::CaseInsensitive) == 0 ||
+	    link.left(8).compare("https://", Qt::CaseInsensitive) == 0) {
```
Comment (JohannMG) [inline UI/frontend-plugins/frontend-tools/scripts.cpp:480]: Might be better to cast into QUrl and use scheme? 
 https://doc.qt.io/qt-5/qurl.html#scheme

File: UI/frontend-plugins/frontend-tools/data/locale/en-US.ini (hunk 1 above)
Comment (JohannMG) [reply UI/frontend-plugins/frontend-tools/data/locale/en-US.ini:42]: Shorter copy similar to browsers and phones.

File: UI/frontend-plugins/frontend-tools/scripts.cpp (hunk 2 above)
Comment (WizardCM) [reply UI/frontend-plugins/frontend-tools/scripts.cpp:480]: Hmm, might be worth doing. I simply copied the code we ended up using for browser source creation via drag and drop.

Comment (WizardCM) [reply UI/frontend-plugins/frontend-tools/scripts.cpp]: Perfect. Done!

File: UI/frontend-plugins/frontend-tools/scripts.cpp [hunk 3]
```diff
@@ -470,6 +474,38 @@ void ScriptsTool::on_defaults_clicked()
...
+			       .arg(link);
+
+		QMessageBox messageBox(this);
+		messageBox.setWindowTitle(
+			obs_module_text("ScriptDescriptionLink.Title"));
+		messageBox.setText(msg);
+
+		obs_frontend_push_ui_translation(obs_module_get_string);
+		QPushButton *yesButton =
+			messageBox.addButton(tr("Yes"), QMessageBox::YesRole);
```
Comment (JohannMG) [inline UI/frontend-plugins/frontend-tools/scripts.cpp:495]: Consider text "Open" i think we already have a translation for it too for Copy/Action matching. See chrome: 
Screen Shot 2020-07-10 at 12 34 46 PM

Comment (jp9000) [reply UI/frontend-plugins/frontend-tools/scripts.cpp]: Alrighty, I swapped it to "Open URL" and "Cancel".

**Pull request merged**